src/mono/mono/mini/interp/interp.c

```diff
@@ -362,6 +362,7 @@ clear_resume_state (ThreadContext *context)
 	g_assert (context->exc_gchandle);
 	mono_gchandle_free_internal (context->exc_gchandle);
 	context->exc_gchandle = 0;
+	context->safepoint_frame = NULL;
```

Comment:
Why null it here ? It should always be null when returning from `mono_threads_safepoint`